src/ansys/geometry/core/shapes/curves/nurbs.py

```diff
@@ -141,6 +142,48 @@ def from_control_points(
 
         return curve
 
+    @classmethod
+    @check_input_types
+    def fit_curve_from_points(
+        cls,
+        points: list[Point3D],
+        degree: int,
+    ) -> "NURBSCurve":
+        """Fit a NURBS curve to a set of points.
+
+        Parameters
+        ----------
+        points : list[Point3D]
+            Points to fit the curve to.
+        degree : int
+            Degree of the curve.
+
+        Returns
+        -------
+        NURBSCurve
+            Fitted NURBS curve.
+
+        """
+        from geomdl import fitting
+
+        # Convert points to a format suitable for the fitting function
+        converted_points = []
+        for pt in points:
+            pt = (pt.x.m, pt.y.m, pt.z.m)
+            converted_points.append(pt)
+
+        # Fit the curve to the points
+        curve = fitting.interpolate_curve(converted_points, degree)
+
+        # Construct the NURBSCurve object
+        nurbs_curve = cls()
+        nurbs_curve._nurbs_curve.degree = curve.degree
+        nurbs_curve._nurbs_curve.ctrlpts = curve.ctrlpts
+        nurbs_curve._nurbs_curve.knotvector = curve.knotvector
+        nurbs_curve._nurbs_curve.weights = curve.weights
+
```

Comment:
```suggestion

        # Verify the curve is valid
        try:
            curve._nurbs_curve._check_variables()
        except ValueError as e:
            raise ValueError(f"Invalid NURBS curve: {e}")

```

Let's do the curve validation just in case...